app/models/service_template_ansible_playbook.rb

```diff
@@ -44,14 +44,20 @@ def self.create_catalog_item(options, auth_user)
     config_info  = validate_config_info(options[:config_info])
 
     enhanced_config = config_info.deep_merge(create_job_templates(service_name, description, config_info, auth_user))
-
-    transaction do
-      create_from_options(options).tap do |service_template|
-        dialog_ids = service_template.send(:create_dialogs, enhanced_config)
-        enhanced_config.deep_merge!(dialog_ids)
-        service_template.options[:config_info].deep_merge!(dialog_ids)
-        service_template.create_resource_actions(enhanced_config)
+    manager_id = enhanced_config.fetch_path(:provision, :configuration_template)[:manager_id]
+
+    begin
+      transaction do
+        create_from_options(options).tap do |service_template|
+          dialog_ids = service_template.send(:create_dialogs, enhanced_config)
+          enhanced_config.deep_merge!(dialog_ids)
+          service_template.options[:config_info].deep_merge!(dialog_ids)
+          service_template.create_resource_actions(enhanced_config)
+        end
       end
+    rescue => e
+      rollback_job_templates(config_info, service_name, manager_id)
+      raise e
```

Comment:
Minor, but if you are not going to use the error object you do not need to assign it to a variable.  Call `raise` without any arguments and it will re-raise the error.

The rescue block would look like this:
```ruby
rescue
  rollback_job_templates(config_info, service_name, manager_id)
  raise
end
```